lib/MetadataStore.js

```diff
@@ -371,6 +371,61 @@ MetadataStore.prototype = {
   }),
 
   /**
+   * Get page metadata (including images) for the given cache_keys.
+   * For the missing cache_keys, it simply ignores them and will not
+   * raise any exception
+   *
+   * @param {Array} cacheKeys an cache key array
+   *
+   * Returns a promise with the array of retrieved metadata records
+   */
+  asyncGetMetadataByCacheKey: Task.async(function*(cacheKeys) {
+    const quoted = cacheKeys.map(key => {return `'${key}'`;}).join(",");
+
+    let metaObjects = yield this.asyncExecuteQuery(
+      `SELECT * FROM page_metadata WHERE cache_key IN (${quoted})`,
```

Comment:
Just curious, did you try comparing the performance of one big query vs many small queries?